tardis/visualization/tools/interaction_radius_plot.py

```diff
@@ -0,0 +1,461 @@
+
+        for data, color, name in zip(
+            plot_data, plot_colors, self._species_name
+            ):
+            hist, _ = np.histogram(data, bins=bin_edges)
+            step_x = np.repeat(bin_edges, 2)[1:-1]
+            step_y = np.repeat(hist, 2)
+            self.ax.plot(step_x, step_y, label=name, color=color, linewidth=2.5, drawstyle='steps-post', alpha=0.75)
+
+        self.ax.ticklabel_format(axis="y", scilimits=(0, 0))
+        self.ax.tick_params("both", labelsize=14)
+        self.ax.set_xlabel("Last Interaction Velocity (km/s)", fontsize=14)
+        self.ax.set_ylabel("Packet Count", fontsize=14)
+        self.ax.grid(True, which='both', linestyle='--', linewidth=0.5)
+        self.ax.legend(fontsize=15)
+        plt.tight_layout()
+
+        return self.ax
+    
+    def generate_plot_plotly(
+        self,
+        packets_mode="virtual",
+        species_list=None,
+    ):
+        """
+        Generate the last interaction radius distribution plot using plotly.
+
+        Parameters
+        ----------
+        packets_mode : str, optional
+            Packet mode, either 'virtual' or 'real'. Default is 'virtual'.
+        species_list : list of str
+            List of species to plot. Default is None.
+
+        Returns
+        -------
+        plotly.graph_objects.Figure
+            Plotly figure object with the plot.
+        """
+        # Parse the requested species list
+        self._parse_species_list(species_list=species_list)
+        species_in_model = np.unique(
+            self.data[packets_mode]
+            .packets_df_line_interaction["last_line_interaction_species"]
+            .values
+        )
+        if self._species_list is None or not self._species_list:
+            raise ValueError("No species provided for plotting.")
+        msk = np.isin(self._species_list, species_in_model)
+        self.species = np.array(self._species_list)[msk]
+        if len(self.species) == 0:
+            raise ValueError("No valid species found for plotting.")
+        # Get the labels in the color bar. This determines the number of unique colors
+        self._make_colorbar_labels()
+        # Get the number of unique colors
+        self._make_colorbar_colors()
+        groups = self.data[packets_mode].packets_df_line_interaction.groupby(
+            by="last_line_interaction_species"
+        )
+
+        plot_colors = []
+        plot_data = []
+
+        for species_counter, identifier in enumerate(self.species):
+            g_df = groups.get_group(identifier)
+            r_last_interaction = g_df["last_interaction_in_r"].values * u.cm
+            v_last_interaction = (r_last_interaction / self.time_explosion).to(
+                "km/s"
+            )
+            plot_data.append(v_last_interaction)
+            color = f"rgba({int(255*self._color_list[species_counter][0])}, {int(255*self._color_list[species_counter][1])}, {int(255*self._color_list[species_counter][2])}, 1)"
```

Comment:
there's `to_rgb255_string` in SDECPlotter, you can use that here and maybe move that staticmethod to plot_util and use in both classes.